### DISCUSSION TITLE
Java extension failed to activate, JDK not working in vscode

### DISCUSSION THREAD

**Discussion by BlessedBiscuit42** [redhat-developer/vscode-java#3328] [Q&A]
I just downloaded vscode and I'm new to coding. I installed the extensions for java online before downloading vscode itself. I made a project and created a file inside. Visual studio code won't let me run or debug. It just keeps giving me the same exact error every single time. I already have a JDK downloaded, but it keeps giving me the error. I deleted and reinstalled the red hat JDK many times, but it won't work. I'm not sure if there is a step I'm missing out or something. I also deleted and reinstalled vscode to try to get it to work, but no luck. I looked at other forums, but I don't understand what they are saying. I don't really understand what is wrong. Could someone please help me? 
![Screenshot 2023-10-01 103651](https://github.com/redhat-developer/vscode-java/assets/146652858/8b5a5c8b-673a-44c0-a6fc-fdabc798c602)
![Screenshot 2023-10-01 103858](https://github.com/redhat-developer/vscode-java/assets/146652858/6c161b34-e9c1-4f11-8a7f-881797d42034)

**Comment by arithauth** [comment]
I am a beginner.

1. Open a new file to test with another workspace. Workspace can be buggy sometimes.

2. Make sure the JDK is properly installed. (JDK 21, which is not yet supported, cannot be used.)
% java -version
java version "20.0.2" 2023-07-18
Java(TM) SE Runtime Environment (build 20.0.2+9-78)
Java HotSpot(TM) 64-Bit Server VM (build 20.0.2+9-78, mixed mode, sharing)

3. If all goes well, this is not necessary, but try setting java.configuration.runtimes. You can get some information by searching "java.configuration.runtimes".

    setting like this:
    "java.configuration.runtimes": [
        { 
            "name": "JavaSE-20",
            "path": "/Library/Java/JavaVirtualMachines/jdk-20.jdk"
        }
    ]

**Reply by rgrunber** [reply]
There's a few things to keep in mind.

- The extension requires a Java runtime (JRE), or a JDK to run and will attempt to locate it in [common locations](https://www.npmjs.com/package/jdk-utils#usage). The easiest way to check is if you can run `java -version` (as mentioned above), then it's on your `PATH` and will likely be found. It seems like your issue might be that you download a JDK and didn't place it in a common location, or add it to `PATH` / `JAVA_HOME` / etc. You can also use [`java.jdt.ls.home`](https://github.com/redhat-developer/vscode-java#java-tooling-jdk) to set the path to the JDK. The extension does require Java 17 to run, but you can use any installed version from 1.5 to 20 for your project.
- If you install the extension from the "Extensions" tab in VS Code, it will detect your platform and install a version that comes pre-packaged with a JDK 17, so you won't even need to download a separate one (unless your projects have separate requirements). See https://github.com/redhat-developer/vscode-java#quick-start .

 Now that I think of it, maybe we should update the README to be more specific about installing from a marketplace, rather than just downloading a vsix (universal) separately, which is what I suspect you tried to do.
